Controller/CoreController.php

```diff
@@ -129,10 +124,27 @@ public function searchAction(Request $request)
         }
 
         return $this->render($this->container->get('sonata.admin.pool')->getTemplate('search'), array(
-            'base_template' => $this->getBaseTemplate($request),
+            'base_template' => $this->getBaseTemplate(),
             'admin_pool'    => $this->container->get('sonata.admin.pool'),
             'query'         => $request->get('q'),
             'groups'        => $this->getAdminPool()->getDashboardGroups(),
         ));
     }
+
+    /**
+     * The purpose of this method it to be BC with symfony 2.
+     *
+     * @deprecated If you use symfony 3, consider using a Request action
+     *             argument.
+     *
+     * @return Request
+     */
+    private function getRequestObject()
```

Comment:
I also struggling sometimes with the consistency vs better code conundrum, and I think being consistent with bad code just creates more bad code. When a better way is found, all new code should use the better way if possible, and bad code should be marked with `FIXME`